staging/src/k8s.io/apiserver/pkg/cel/library/network_addrs.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"net/netip"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// NetworkAddr provides a CEL function library extension of network address parsing functions.
+//
+// networkAddr
+//
+// Converts a string to a network address or results in an error if the string is not a valid network address.
+// The network address must be an IPv4 or IPv6 address with a CIDR mask.
+//
+//	networkAddr(<string>) <NetworkAddr>
+//
+// Examples:
+//
+//	networkAddr('192.168.0.0/16') // returns an IPv4 address with a CIDR mask
+//	networkAddr('::1/128') // returns an IPv6 address with a CIDR mask
+//	networkAddr('192.168.0.0/33') // error
+//	networkAddr('::1/129') // error
+//
+// isNetworkAddr
+//
+// Returns true if a string is a valid network address. The network address must be an IPv4 or IPv6 address with a CIDR mask.
+//
+//	isNetworkAddr(<string>) <bool>
+//
+// Examples:
+//
+//	isNetworkAddr('192.168.0.0/16') // returns true
+//	isNetworkAddr('::1/128') // returns true
+//	isNetworkAddr('192.168.0.0/33') // returns false
+//	isNetworkAddr('::1/129') // returns false
+//
+// containsIPAddr
+//
+// - containsIPAddr: Returns true if a network address contains an IP address. Should be passed an IP address parsed with ipAddr.
+//
+// Examples:
+//
+// networkAddr('192.168.0.0/24').containsIPAddr(ipAddr('192.168.0.1')) // returns true
+// networkAddr('192.168.0.0/24').containsIPAddr(ipAddr('192.168.1.1')) // returns false
```

Comment:
This syntax seems like it will _always_ require you to call `ipAddr()` on the argument to `containsIPAddr` wouldn't it? It would be simpler to just have `containsIPAddr` take a string

Reply:
Yeah it does, this was inspired by other CEL extension like the quantity CEL that has a similar pattern. I'll let @cici37 chime in here, I think this is the expected pattern, but if taking a string is simpler then we can do that too